### ISSUE TITLE
Go-to-definition is not triggered

### ISSUE THREAD

**Issue by InSyncWithFoo** [InSyncWithFoo/pyright-langserver-for-pycharm#29]
See #26.

**Comment by InSyncWithFoo** [comment]
It seems that PyCharm always sends a `textDocument/codeAction` instead of a `textDocument/definition` (to which Pyright always replies with `{"jsonrpc":"2.0","id":"0","result":[]}`):

```json
{
	"jsonrpc": "2.0",
	"id": "0",
	"method": "textDocument/codeAction",
	"params": {
		"textDocument": {
			"uri": "file:///<project>/foo.py"
		},
		"range": {
			"start": { "line": 1, "character": 4 },
			"end": { "line": 1, "character": 4 }
		},
		"context": {
			"diagnostics": [],
			"triggerKind": 2
		}
	}
}
```

To me, this screams "Bug" and I would be very surprised if it isn't one.

I have asked about this [on Slack](https://jetbrains-platform.slack.com/archives/C5U8BM1MK/p1714149417088039). If there are no replies in a few days, I will report it to YouTrack.

**Comment by InSyncWithFoo** [comment]
Upstream issue: [IDEA-352715](https://youtrack.jetbrains.com/issue/IDEA-352715).

**Comment by InSyncWithFoo** [comment]
Upstream issue declined:

> IntelliJ's LSP API is designed first of all to provide _additional functionality_, but not to replace existing functionality. The main use case for LSP-based external plugins is to add support for programming languages that are not supported natively by JetBrains IDEs.
> 
> Adding features to the programming languages supported by JetBrains IDEs is also supported by the LSP API. But in this case, some limitations may apply. These limitations are not intentional, but technical.
> 
> Like your use case. It's not that we intentionally break external plugin functionality. But it's tricky for us to make it work. And since such a use case (replacing references in Python files) is not the primary goal of the LSP API, we can't promise a fix.
